### ISSUE TITLE
Override GenerateNuspec or other targets in Directory.Build.targets

### ISSUE THREAD

**Issue by TFTomSun** [microsoft/msbuild#4747]
I expected that Directory.Build.targets is imported at the end of the Microsoft.NET.Sdk. Unfortunately some targets seem to be defined after the directory.build.targets has been imported, which prevents me from overriding them. Is there a property to a target file that is imported after everything else is already specified? Is there any other possibility to override targets like GenerateNuspec without changing the standard project file format (with Sdk attribute in the project tag)?

**Comment by TFTomSun** [comment]
I investigated a bit further... I think the problem is that the nuget build task sdk is imported after the microsoft.net.sdk.targets

```xml

 <Import Project="$(MSBuildThisFileDirectory)..\targets\Microsoft.NET.Sdk.targets"
          Condition="'$(IsCrossTargetingBuild)' != 'true'"/>
  
  <!-- Import targets from NuGet.Build.Tasks.Pack package/Sdk -->
  <PropertyGroup Condition="'$(NuGetBuildTasksPackTargets)' == '' AND '$(ImportNuGetBuildTasksPackTargetsFromSdk)' != 'false'">
    <NuGetBuildTasksPackTargets Condition="'$(IsCrossTargetingBuild)' == 'true'">$(MSBuildThisFileDirectory)..\..\NuGet.Build.Tasks.Pack\buildCrossTargeting\NuGet.Build.Tasks.Pack.targets</NuGetBuildTasksPackTargets>
    <NuGetBuildTasksPackTargets Condition="'$(IsCrossTargetingBuild)' != 'true'">$(MSBuildThisFileDirectory)..\..\NuGet.Build.Tasks.Pack\build\NuGet.Build.Tasks.Pack.targets</NuGetBuildTasksPackTargets>
    <ImportNuGetBuildTasksPackTargetsFromSdk>true</ImportNuGetBuildTasksPackTargetsFromSdk>
  </PropertyGroup>
  
  <Import Project="$(NuGetBuildTasksPackTargets)"
          Condition="Exists('$(NuGetBuildTasksPackTargets)') AND '$(ImportNuGetBuildTasksPackTargetsFromSdk)' == 'true'"/>  

```

**Comment by livarcocc** [comment]
The issue is that the thing that imports the NuGet targets IS the Microsoft.Net.Sdk.

**Comment by japj** [comment]
The only thing that might be useful is to hook into GenerateNuspecDependsOn and possibly modify the inputs for GenerateNuspec?

**Comment by TFTomSun** [comment]
@japj yeah maybe, but that doesn't allow me to override the original functionality, just let me specify the parameters in a way that the original functionality does nothing. I'll consider it as a workaround.

**Comment by TFTomSun** [comment]
@livarcocc I think if the second import comes before the first one, it should solve the issue, shouldn't it?

**Comment by japj** [comment]
You could also look into using ImportNuGetBuildTasksPackTargetsFromSdk false and import those from your own targets and overrule afterwards

**Comment by austinmfb** [comment]
I am having a similar issue with dotnet pack ignoring Directory.Build.props for embedding debug symbols in my case. If I am just building and packaging c# libraries, why would the build phase ignore this file?
